### PR TITLE
Fixes #1699 - Introduce a new advanced settings screen

### LINKED ISSUE #1699: Show advanced settings for external users
### Story

As an external user that wants to configure their element client beyond the default setting I would love to access certain feature flags in a separated settings view.

- Create an "Advanced settings view" for external users. In the settings view.

The following options should be configurable:
- Message Layout (Bubbles vs. Modern)
-- Use the existing setting and move it here
- Enable Rich Text Editor
- - Use the existing setting from developer options and move it here @langleyd Good idea or should this be an other one?
- View event source option
- - Needs to be build. Enables to see the event source option on long pressing an event.

### PR DESCRIPTION
**Pull request by stefanceriu**
Screenshot 2023-09-13 at 16 52 45

Screenshot 2023-09-13 at 16 52 47

### CODE REVIEW

File: ElementX/Sources/Screens/Settings/AvancedOptionsScreen/View/AdvancedSettingsScreen.swift [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+            Section {
+                ListRow(label: .default(title: L10n.commonMessageLayout,
+                                        systemIcon: .rectangleGrid1x2),
+                        kind: .picker(selection: $context.timelineStyle,
+                                      items: TimelineStyle.allCases.map { (title: $0.name, tag: $0) }))
+                
+                ListRow(label: .default(title: L10n.actionViewSource, systemIcon: .mailAndTextMagnifyingglass),
+                        kind: .toggle($context.viewSourceEnabled))
+                
+                ListRow(label: .default(title: L10n.commonRichTextEditor, systemIcon: .squareFillTextGrid1x2),
```
Comment (pixlwave) [inline ElementX/Sources/Screens/Settings/AvancedOptionsScreen/View/AdvancedSettingsScreen.swift:35]: Seeing this symbol name, I think its more semantically intended for e.g. a list of articles with a thumbnail. Maybe we should use one of the `list…` symbols instead? But probably more a question for @amshakal.

File: ElementX/Sources/Screens/Settings/SettingsScreen/View/SettingsScreen.swift [hunk 2]
```diff
@@ -113,48 +98,52 @@ struct SettingsScreen: View {
...
+            
+            if context.viewState.showDeveloperOptions {
+                ListRow(label: .default(title: L10n.commonDeveloperOptions,
+                                        systemIcon: .hammerCircle),
+                        kind: .navigationLink {
+                            context.send(viewAction: .developerOptions)
+                        })
+                        .accessibilityIdentifier(A11yIdentifiers.settingsScreen.developerOptions)
+            }
+        }
```
Comment (pixlwave) [inline ElementX/Sources/Screens/Settings/SettingsScreen/View/SettingsScreen.swift:146]: This shouldn't be inside `simplifiedSection`, can we extract it into its own property like the others?

**Review by pixlwave — APPROVED** [review]
Nice 👍

File: ElementX/Sources/Screens/Settings/AvancedOptionsScreen/View/AdvancedSettingsScreen.swift (hunk 1 above)
Comment (stefanceriu) [reply ElementX/Sources/Screens/Settings/AvancedOptionsScreen/View/AdvancedSettingsScreen.swift]: It's okay, we replaced it with the same icon we use in the attachments menu

File: ElementX/Sources/Screens/Settings/SettingsScreen/View/SettingsScreen.swift (hunk 2 above)
Comment (stefanceriu) [reply ElementX/Sources/Screens/Settings/SettingsScreen/View/SettingsScreen.swift]: Ended up moving both to an `advancedOptionsSection`

File: ElementX/Sources/Screens/Settings/AvancedOptionsScreen/View/AdvancedSettingsScreen.swift [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    var body: some View {
+        Form {
+            Section {
+                ListRow(label: .default(title: L10n.commonMessageLayout,
+                                        systemIcon: .rectangleGrid1x2),
+                        kind: .picker(selection: $context.timelineStyle,
+                                      items: TimelineStyle.allCases.map { (title: $0.name, tag: $0) }))
+                
+                ListRow(label: .default(title: L10n.actionViewSource, systemIcon: .mailAndTextMagnifyingglass),
+                        kind: .toggle($context.viewSourceEnabled))
```
[the remark this answers is not in the document]
Comment (stefanceriu) [reply ElementX/Sources/Screens/Settings/AvancedOptionsScreen/View/AdvancedSettingsScreen.swift:33]: We could but I would prefer to leave it alone

**Comment by bmarty** [comment]
"View Source" wording is maybe a bit confusing here. Maybe something like "Enable View Source action", or "Be able to view the source of Event" would fit better. Not sure what is best (probably something else!). On Element Android, there is no such option, but when "Developer mode" is on, this action become visible. It is a bit better I think...

image

**Comment by amshakal** [comment]
LGTM

**Review by amshakal — APPROVED** [review]

**Pull request merged**